pkg/controllers/nodeclaim/termination/suite_test.go

```diff
@@ -155,47 +159,15 @@ var _ = Describe("Termination", func() {
 		ExpectNotFound(ctx, env.Client, node1, node2, node3)
 
 		ExpectObjectReconciled(ctx, env.Client, nodeClaimTerminationController, nodeClaim) // now all nodes are gone so nodeClaim deletion continues
+
+		ExpectMetricHistogramSampleCountValue("karpenter_nodeclaims_instance_termination_duration_seconds", 1, map[string]string{"nodepool": nodePool.Name})
+		ExpectMetricHistogramSampleCountValue("karpenter_nodeclaims_termination_duration_seconds", 1, map[string]string{"nodepool": nodePool.Name})
 		ExpectNotFound(ctx, env.Client, nodeClaim, node1, node2, node3)
 
 		// Expect the nodeClaim to be gone from the cloudprovider
 		_, err = cloudProvider.Get(ctx, nodeClaim.Status.ProviderID)
 		Expect(cloudprovider.IsNodeClaimNotFoundError(err)).To(BeTrue())
 	})
-	It("should not delete the NodeClaim until all the Nodes are removed", func() {
```

Comment:
Why did we remove this test?

Reply:
This test seemed valid to me when we were doing this as part of the nodeClaim termination controller -

if err = c.kubeClient.Delete(ctx, node); client.IgnoreNotFound(err) != nil {
				return reconcile.Result{}, err
}
// We wait until all the nodes associated with this nodeClaim have completed their deletion before triggering the finalization of the nodeClaim
if len(nodes) > 0 {
return reconcile.Result{}, nil
}

I removed the `len(nodes)>0` check because we discussed that we want to make `cloudProvider.Delete()` atleast once in the termination flow.
I can add this test and tweak it to check that we are not removing finalizer on nodeClaim until node is gone.